### ISSUE TITLE
Version mismatch between worker/scheduler and notebook in images from docker hub

### ISSUE THREAD

**Issue by soroosh-rzh** [dask/dask-docker#161]
I pulled the latest images of `daskdev/dask` and `daskdev/dask-notebook` from Docker hub. It sounds like there is a version mismatch between them

![dask-error](https://user-images.githubusercontent.com/62025982/123166457-6eb22c80-d443-11eb-9659-e6e37d3ceca1.png)

Moving forward to investigate my dask dataframe it created error. I'm new to dask and not sure this version mismatch is really the root cause but that's what error message complained!

**Comment by jrbourbeau** [comment]
I transferred this issue over to `dask-docker` as that's where the source for these images live. I'll second @jsignell's comment that more information about the error you encountered would be useful. What images are you using for the client, scheduler, and workers, respectively?

**Comment by soroosh-rzh** [comment]
I just pulled all the images from docker hub today (latest tag)
I attached the error
[errortrace.txt](https://github.com/dask/dask-docker/files/6705034/errortrace.txt)

**Comment by jrbourbeau** [comment]
How are you setting up you cluster? And inside which docker image?

**Comment by lbonini94** [comment]
I made a PR to fix this: https://github.com/dask/dask-docker/pull/162

**Comment by jacobtomlinson** [comment]
A reasonable amount has changed in #166 and #149, particularly the base images, so I wanted to circle back here and see what the state of things are.

It seems like there are still some minor differences between the images when following the example in the README.

<img width="963" alt="image" src="https://user-images.githubusercontent.com/1610850/129715795-cf760502-932a-49a3-b7a9-76bd9ed2e37f.png">

**Comment by jacobtomlinson** [comment]
Looks like #173 has resolved the Python mismatch.

<img width="937" alt="image" src="https://user-images.githubusercontent.com/1610850/129718341-38624875-285e-417f-a936-98c641037754.png">

Strangely we are still seeing a `blosc` mismatch despite `blosc` being listed in the install command for the base notebook.

https://github.com/dask/dask-docker/blob/c3e51643a0f3fed407a89d493de12ade7b217bc8/base/Dockerfile#L16

However if I run the container and try to install `blosc` with either `conda` or `mamba` I get `All requested packages already installed`. But I am unable to `import blosc`.

```python
>>> import blosc
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ModuleNotFoundError: No module named 'blosc'
```

@jrbourbeau do you have any ideas?

**Comment by jacobtomlinson** [comment]
Ah looks like I want `python-blosc`. Let me try again.

**Comment by jacobtomlinson** [comment]
Woo! #174 did it.

<img width="349" alt="image" src="https://user-images.githubusercontent.com/1610850/129729004-8a5a14bc-83e1-4923-a3e0-6b31dd03648d.png">

**Comment by italovinicius18** [comment]
@jacobtomlinson I'm getting a similar error:

![image](https://github.com/dask/dask-docker/assets/39673124/0316f5a1-58e4-4417-8548-c28325d97850)

I'm reading a sql table, when I make the method head on the dataframe, this occurs:

![image](https://github.com/dask/dask-docker/assets/39673124/aef2b1fb-b40f-4e79-a18f-b3804afd72e1)

**Comment by jacobtomlinson** [comment]
@italovinicius18 it looks like your local environment has Python 3.10.11 and the scheduler/worker container has 3.10.10. Are you using the notebook container image for your client? I'm surprised such a minor version difference results in a deserialization error like that though.

Could you share a Minimal Reproducible Example with your environment and the code you are running so we can try and reproduce?

**Comment by italovinicius18** [comment]
Hello, Jacob!

I was using the latest version, but I fixed changing the notebook, scheduler and workers to 2023.6.0-py3.9.

Check my current docker:


```yml
version: "3.1"

services:
  scheduler:
    image: ghcr.io/dask/dask:2023.6.0-py3.9
    hostname: scheduler
    ports:
      - "8786:8786"
      - "8787:8787"
    command: ["dask-scheduler"]
    environment:
      - EXTRA_PIP_PACKAGES=psycopg2-binary SQLAlchemy hvplot

  worker:
    image: ghcr.io/dask/dask:2023.6.0-py3.9
    command: ["dask-worker", "tcp://scheduler:8786"]
    # For Docker swarm you can specify multiple workers, this is ignored by `docker-compose up`
    deploy:
      replicas: 2
    environment:
      - EXTRA_PIP_PACKAGES=psycopg2-binary SQLAlchemy hvplot

  notebook:
    image: ghcr.io/dask/dask-notebook:2023.6.0-py3.9
    ports:
      - "8888:8888"
    environment:
      - DASK_SCHEDULER_ADDRESS="tcp://scheduler:8786"
      - EXTRA_PIP_PACKAGES=psycopg2-binary SQLAlchemy hvplot
    volumes:
      - ./work:/home/jovyan/work

  db-dask:
    container_name: db-news-dask
    image: postgres:13
    ports:
      - 5432:5432
    volumes:
      - ../producer/data:/var/lib/postgresql/data
    environment:
      POSTGRES_USER: news
      POSTGRES_PASSWORD: news
      POSTGRES_DB: news_db
    healthcheck:
      test: ["CMD-SHELL", "pg_isready -U news -d news_db -p 5432"]
      interval: 10s
      timeout: 5s
      retries: 5
```

**Comment by jacobtomlinson** [comment]
Were you using `latest` before?

**Comment by italovinicius18** [comment]
Yes

**Comment by jacobtomlinson** [comment]
Could you try again with the `2023.6.1` images and see if the problem still persists?
